### ISSUE TITLE
Feature request : forced scanlines level across all cores for forced scandoubler=1

### ISSUE THREAD

**Issue by stainmage** [MiSTer-devel/Main_MiSTer#777]
Hello,

I think this would be useful if we want to the scandoubler to be set to on + vga scanlines for all cores on one ini preset

Cheers

**Comment by sorgelig** [comment]
There is no one specific type of scanlines in MiSTer. It also may depend on core and other options. So it's per core setting which is not hard to set once and save.

**Comment by stainmage** [comment]
Hello sorgelig,

First thanks for your answer and for your work on the project.

I can give you a practical case : I hook the mister on a vga monitor without scanlines, which needs scandoubler

Forced scandoubler makes things simple all around.

However setting scanlines on each cores means I have to enter each core individually, and set it, it's ok for consoles but arcade cores are plenty.

This means I need to take something like 3-4 hours to enter every core and set the scanlines. It's very tedious operation, and not so simple to revert because I need to delete every cfg file after that or revert if I want to change it.

There are 4 types of scanlines , 25 50 75 and HQ, which means 4 values. So It would be really easier for me to set something like scanlines=1 (for 25 let's say) and having every core set to that value like every core forced to scandoubling.

Or maybe it can be forced scandoubler=1 without scanline, 2 with 25%, 4 with HQ...

It's time saver to be able to do that from the main file.

I know it's not hard to set one core but too many arcade cores around.

**Comment by birdybro** [comment]
At least for one part of your request regarding arcade cores, the functionality is already there. Add this to the end of your MiSTer.ini for the arcade cores. You can also specify the core names in the brackets like this for any other core.

```ini
[arcade]
forced_scandoubler=1
```

**Comment by stainmage** [comment]
yes this is very useful, I wish we could complete this function with more precision on which scandoubler mode is used

**Comment by sorgelig** [comment]
scandoubler and scanlines are different options by nature. forced_scandoubler option is in ini because you cannot switch it in OSD on incompatible monitor. 
scanlines are per-core decision. Not all cores support scanlines, and it may depends on resolution core outputs. Also OSD options cannot be controlled from OSD and ini at the same time.

**Comment by thorr2** [comment]
I was going to propose an idea for a feature request, but this issue is pretty close already, so I will just propose it here.  The forced_scandoubler=1 is great for running cores with zero lag on a VGA monitor.  It draws each line twice progressively down the screen so it can work with the higher frequency VGA monitor.  My proposal is to offer another option of forced_scandoubler=2 that draws a black line for the second line instead of repeating the same line twice.

**Comment by sorgelig** [comment]
Basically i've already answered question above. forced_scandoubler option is in ini because it cannot be switched in OSD if monitor doesn't support 15KHz. Options for scanlines are controlled by each core and not every core may have such option. Don't be lazy, enable it in core when you start to use specific core. It needs to be done once per core. INI is for other purposes.
